util/caller/resolver.go

```diff
@@ -55,57 +69,39 @@ func NewCallResolver(offset int, numStrip int) *CallResolver {
 		panic("could not look up callsite of NewCallResolver")
 	}
 	fileParts := strings.Split(file, "/")
-	cr.stripPrefix = strings.Join(fileParts[:len(fileParts)-numStrip-1], "/") + "/"
+	if len(fileParts) > numStrip {
+		cr.stripPrefix = strings.Join(fileParts[:len(fileParts)-numStrip-1], "/") + "/"
```

Comment:
Done: using a regex, with the default instance dropping the project root (`.../cockroachdb/cockroach/` or `.../coreos/etcd/`).